src/Faction.js

```diff
@@ -306,13 +306,16 @@ function displayFactionContent(factionName) {
     var donateDivWrapper = createElement("div", {class:"faction-work-div-wrapper"});
     donateDiv.appendChild(donateDivWrapper);
     var donateRepGain = createElement("p", {
-        innerText:"This donation will result in 0 reputation gain"
+        innerText:"This donation will result in 0.000 reputation gain"
     });
     var donateAmountInput = createElement("input", {
         placeholder:"Donation amount",
         inputListener:()=>{
-            var amt = parseFloat(donateAmountInput.value);
-            if (isNaN(amt) || amt < 0) {
+            let amt = 0;
+            if(donateAmountInput.value !== "") {
+                amt = parseFloat(donateAmountInput.value);
```

Comment:
I wonder if we really want something more along the lines of the example function under the ["A stricter parse function" section from MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseFloat#A_stricter_parse_function) to allow for entering amounts like "1M" or "1B"